cdc_rsync_server/cdc_rsync_server.cc

```diff
@@ -619,27 +726,25 @@ absl::Status CdcRsyncServer::HandleSendMissingFileData() {
     }
 
     status = path::StreamWriteFileContents(*fp, handler);
-    fclose(*fp);
-    if (!status.ok()) {
-      return WrapStatus(status, "Failed to write file %s", filepath);
-    }
+    finalize_pool.QueueTask(std::make_unique<FinalizeCopiedFileTask>(
+        *fp, file, filepath, is_executable, status));
+    finalize_pool.WaitForQueuedTasksAtMost(kMaxQueueSize);
 
-    // Set file write time.
-    status = path::SetFileTime(filepath, file.modified_time);
-    if (!status.ok()) {
-      return WrapStatus(status, "Failed to set file mod time for %s", filepath);
+    // Drain finalize pool for the last file.
+    if (server_index + 1 == diff_.missing_files.size()) {
```

Comment:
Nit: The name `server_index` is confusing to me. It took me a while to figure out that it is the index into `missing_files`. Since this is technically not related to this change, feel free to do with that information what you want. :-)

Reply:
The terms `client_index` and `server_index` are used throughout the sync code. The former is an index into arrays of files on the client, whereas the latter is the equivalent for the server. Some code in the client deals with both at the same time, and this naming scheme keeps the code sane.